undercloud.yml

```diff
@@ -2,28 +2,37 @@
   vars:
     container_params: ""
   tasks:
-      - name: get undercloud_local_interface
-        shell: |
-          for i in /sys/class/net/*
-          do
-            udevcontent=`udevadm info -p $i --query property`
-            if [[ $udevcontent =~ {{ ctlplane_interface }} ]]
-            then
-              undercloud_local_interface=`echo $i | cut -d '/' -f 5`
-              break
-            fi
-          done
-          echo $undercloud_local_interface
-        register: local_interface
-        delegate_to: "{{ undercloud_hostname }}"
-        vars:
-          ansible_python_interpreter: "{{ python_interpreter }}"
-          ansible_user: "stack"
-        become: true
+      - name: set undercloud local interface
+        block:
+          - name: get undercloud_local_interface
+            shell: |
+              for i in /sys/class/net/*
+              do
+                udevcontent=`udevadm info -p $i --query property`
+                if [[ $udevcontent =~ {{ ctlplane_interface }} ]]
+                then
+                  undercloud_local_interface=`echo $i | cut -d '/' -f 5`
+                  break
+                fi
+              done
+              echo $undercloud_local_interface
+            register: local_interface
+            delegate_to: "{{ undercloud_hostname }}"
+            vars:
+              ansible_python_interpreter: "{{ python_interpreter }}"
+              ansible_user: "stack"
+            become: true
+
+          - name: set undercloud_local_interface
+            set_fact:
+              undrecloud_local_interface: "{{ local_interface.stdout }}"
+        when: virtual_uc != true
 
       - name: set undercloud_local_interface
         set_fact:
-          undrecloud_local_interface: "{{ local_interface.stdout }}"
+          undrecloud_local_interface: "eth0"
```

Comment:
Need to test with RHEL 8 image if it uses eth* naming convention. But for now it is fine. Please use variable.

Reply:
done